### ISSUE TITLE
Issue with saving automod rules.

### ISSUE THREAD

**Issue by PunisherDev** [OutwalkStudios/discord-backup#14]
```javascript
TypeError: Cannot read properties of undefined (reading 'id')
    at /home/container/node_modules/@outwalk/discord-backup/dist/index.js:1:8225
    at /home/container/node_modules/@discordjs/collection/dist/index.js:242:14
    at Function.from (<anonymous>)
    at _Collection.map (/home/container/node_modules/@discordjs/collection/dist/index.js:240:18)
    at /home/container/node_modules/@outwalk/discord-backup/dist/index.js:1:8211
    at _Collection.forEach (<anonymous>)
    at Object.A [as getAutoModerationRules] (/home/container/node_modules/@outwalk/discord-backup/dist/index.js:1:7769)
    at async Object.X [as create] (/home/container/node_modules/@outwalk/discord-backup/dist/index.js:2:54)
```

**Comment by jleeson** [comment]
Hi @PunisherDev, can you provide some code to help create a minimal reproduction of this error?

**Comment by PunisherDev** [comment]
Code definitely isn't the issue because it works. Just servers with automod rules cause issues.

```javascript
                    const data = await Backup.create(interaction.guild, {
                        backupId: new Date().getTime().toString(),
                        maxMessagesPerChannel: messages,
                        jsonBeautify: true,
                        jsonSave: false,
                        backupMembers: false,
                        saveImages: "url",
                        speed: 250
                    })
```

**Comment by jleeson** [comment]
@PunisherDev, I agree this is more than likely some edge case in the getAutoModerationRules function in this package, I did some testing using the snippet you provided to ensure the same settings are used and while using a variety of auto moderation rules, I am unable to reproduce the error you are seeing. I did however discover that due to a typo exempt channels are not restored from a backup.

Can you provide a screenshot of the automod rules you are getting the error from? the only use of an "id" property in the function mentioned by the error is for exempt roles and channels, but I am currently unable to get that to result in an error.

**Comment by SCDerox** [comment]
Hi, we've been experiencing this issue with a small percentage of servers.

Here's a screenshot of one of the auto-mod configuration:
![image](https://github.com/OutwalkStudios/discord-backup/assets/43812987/1c0e6a90-cd5f-4cdf-9579-adfbd929c3e7)
And [here](https://gist.github.com/SCDerox/971bbb20e603b51fc7b2c5efa350209f)'s the output of `JSON.stringify((await guild.autoModerationRules.fetch()))`.

I've been trying to diagnose this issue myself, but no luck so far, I hope this helps you find the bug. This does not happen to all servers with Auto-Mod enabled and I'm unable to find anything that is common with all affected servers.

**Comment by SCDerox** [comment]
Sorry for the confusion, this is caused by the way JSON.stringify works with discord.js's collections, that's my mistake - but I was able to reproduce the issue: 
1. Create a new channel and add it to any automod rule as an exempted channel) 
2. Delete the newly created channel (after saving the automod rule).
3. Retry the backup. 
This will cause the deleted channel to be `undefined`. Logging guild.autoModerationRules.fetch():
![image](https://github.com/OutwalkStudios/discord-backup/assets/43812987/603e2d95-ee19-46e0-bc46-1f630e5c3a1c)
4. Observe how the backups failes, resulting in the error message "Cannot read properties of undefined (reading 'id')".

I'm not quite sure if this is something discord.js is intending, but for now, it would be possible to filter for undefined values in [src/functions/create.js 190](https://github.com/OutwalkStudios/discord-backup/blob/53d48d3f2aa85b3c36878055286cee5b8a9bc59c/src/functions/create.js#L160).

**Comment by jleeson** [comment]
Great catch, I had a feeling it was the result of deleting a channel or role, but i was unable to reproduce it. I didn't want to filter out undefined if there was a bug in this package causing the undefined in the first place. Ill get the filter added and push an update here in a minute.

**Comment by jleeson** [comment]
@SCDerox @PunisherDev this bug should now be resolved in v0.5.4 of this package. If for some reason the problem persist after updating, I will reopen this issue and continue investigating.
